Remotion/Validation/Core/Attributes/Validation/NotEmptyValidationAttribute.cs

```diff
@@ -38,18 +38,24 @@ protected override IEnumerable<IPropertyValidator> GetValidators (IPropertyInfor
       ArgumentUtility.CheckNotNull("property", property);
       ArgumentUtility.CheckNotNull("validationMessageFactory", validationMessageFactory);
 
-      NotEmptyValidator validator;
-      if (string.IsNullOrEmpty(ErrorMessage))
+      IPropertyValidator CreateValidator (Func<ValidationMessage, IPropertyValidator> factory)
```

Comment:
You skipped the question - I would like to qualify the pros and cons of the local-function vs member method decision.

Reply:
Sorry, I did not see the posting starting with "Question", only that above.
Converting the local function to a member method adds syntactic noise, because I have to pass two additional parameters in all 5 lines.
The most important information in this method is the mapping between `property.PropertyType` and the `factory` delegate. I could put that in a dictionary to make it explicit, and have the rest of the code looking up whether an entry exists and creating the `IPropertyValidator` from the entry's `factory` delegate.
What do you think?

Reply:
Ah, I see. That's actually an interesting idea! Bascially, we test for the type and use that as the key. We won't get rid of the if/else block because we still need to calculate the key based on the PropertyType though and I'm not sure if that works well. Perhaps just extract the "get a delegate" part not in a dictionary but in a separate code block that does just the mapping of PropertyType to delegate and then go from there. This could then be a separate method, perhaps just a local function or just a code block - not sure yet.

``` 
  var factory;
  if (type == string)
    factory = msg=> new NotEmptyStringValidator(msg);
  /* other conditions */
  else if (type.IsAssignableTo(ICollection<>)
    factory = msg => ...
  else
    throw ....
```

Also, noticed, we don't need `IsAsignable` with `string` and `byte[]`, just equality. I completely spaced on that while reading the code.